src/main/java/com/next/step/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,60 @@
+package com.next.step.step2;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final int NULL_EMPTY_RETURN_VALUE = 0;
+
+    private static final Pattern CUSTOM_PATTERN = Pattern.compile("//(.)\n(.*)");
+
+    private static final String DEFAULT_DELIMITER = "[,:]";
+
+    private static final int DELIMITER_EXTRACT_NUMBER = 1;
+
+    private static final int INPUT_EXTRACT_NUMBER = 2;
+
+    public static int splitAndSum(String input) {
+        if (isNullOrEmpty(input)) {
+            return NULL_EMPTY_RETURN_VALUE;
+        }
+
+        return addSplitInput(splitInput(input));
+    }
+
+    private static boolean isNullOrEmpty(String input) {
+        return input == null || input.isEmpty();
+    }
+
+    private static String[] splitInput(String input) {
+        Matcher matchedValue = CUSTOM_PATTERN.matcher(input);
+        if (isMatched(matchedValue)) {
+            return extractSplitInput(matchedValue);
+        }
+        return input.split(DEFAULT_DELIMITER);
+    }
```

Comment:
```suggestion
    private static String[] split(String input) {
        Matcher customPatternMatcher = CUSTOM_PATTERN.matcher(input);
        if (customPatternMatcher.find()) {
            String delimiter = customPatternMatcher.group(DELIMITER_EXTRACT_NUMBER);
            return customPatternMatcher.group(INPUT_EXTRACT_NUMBER).split(delimiter);
        }
        return input.split(DEFAULT_DELIMITER);
    }
```

- 변수의 이름 짓기가 참 어렵죠.. 이 경우엔 custom pattern의 matcher 인스턴스의 변수명이니 그냥 위와 같이 지을거 같네요. 
- 메소드를 극단적으로 작게 나누는 것을 이번 과정에서 계속 요구하긴 할텐데요. 저는 위의 경우에는 저정도 수준에서 둘 거 같아요. 메소드를 나눌 때 추상화 수준을 맞추려는 편인데요. split 메소드를 그대로 쓰고, 상수를 노출 시켰듯 if 조건 분기 내에서도 메소드로 분리할 필요가 있나 하는 생각이 드네요. 그리고 메소드 이름도 직관적으로 와닿지는 않구요.
- 저는 메소드 이름에서 인자와 같은 단어를 사용하는걸 지양하고 있어요. (저것도 중복이라..)

Reply:
네. 극단적으로 나누는 것보다 노출시키는게 나을 것 같네요. 하지만 메서드 이름에 인자와 같은 단어를 사용하는 것이 중복이라는 건 조금 의견이 달라서요. 

* 중복이라면 중복일 수 있지만 인자 및 메서드 포함명이 서로 다르게되면 더 혼선을 가져올 수 있다고 생각해요.
* 그리고 동사로만 표현하기엔 자바에서 제공하는 메서드랑 헷갈릴 수 있는 부분도 많아서요.

그래서 메서드 네임은 이름은 그대로 가져가려구 합니다:)